src/test/java/org/galatea/starter/entrypoint/IexRestControllerTest.java

```diff
@@ -81,4 +81,142 @@ public void testGetLastTradedPriceEmpty() throws Exception {
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesNoRange() throws  Exception {
+    MvcResult result = this.mvc.perform(
+            org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+                .get("/iex/historicalPrices?symbol=FB&date=20200409")
+                .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("175.71")))
+        .andExpect(jsonPath("$[1].low").value(new BigDecimal("174.365")))
+        .andExpect(jsonPath("$[2].open").value(new BigDecimal("174.8")))
+        .andExpect(jsonPath("$[3].close").value(new BigDecimal("174.55")))
+        .andExpect(jsonPath("$[5].volume").value(new BigDecimal(1155)))
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesNoSymbol() throws Exception {
+    MvcResult result = this.mvc.perform(
+        org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+            .get("/iex/historicalPrices?symbol=")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isNotFound())
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesRangeAndDate() throws Exception {
+    MvcResult result = this.mvc.perform(
+        org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+            .get("/iex/historicalPrices?symbol=&range=1m&date=20200409")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isNotFound())
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesDateandSymbol() throws Exception {
+    MvcResult result = this.mvc.perform(
+        org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+            .get("/iex/historicalPrices?symbol=FB&range=&date=20200409")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("175.71")))
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesSymbolOnly() throws Exception {
+    MvcResult result = this.mvc.perform(
+        org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+            .get("/iex/historicalPrices?symbol=FB&range=&date=")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("175.71")))
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesRangeOnly() throws Exception {
+    MvcResult result = this.mvc.perform(
+            org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+                .get("/iex/historicalPrices?symbol=&range=1m&date=")
+                .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("175.71")))
+        .andReturn();
```

Comment:
Providing just a range shouldn't return a value, right? You should add more mocked JSON responses to reflect the actual behavior of the IEX server in this scenario.

Reply:
Ya I just goofed here sorry